activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-jpa/src/main/java/org/activiti/cloud/services/modeling/jpa/ModelRepositoryImpl.java

```diff
@@ -108,7 +108,7 @@ public ModelEntity createModel(ModelEntity model) {
     public ModelEntity updateModel(ModelEntity modelToBeUpdated, ModelEntity newModel) {
         Optional.ofNullable(newModel.getName()).ifPresent(modelToBeUpdated::setName);
         Optional.ofNullable(newModel.getExtensions()).ifPresent(modelToBeUpdated::setExtensions);
-        versionGenerationHelper.generateNextVersion(modelToBeUpdated);
+        if (modelToBeUpdated != newModel) versionGenerationHelper.generateNextVersion(modelToBeUpdated);
```

Comment:
Given the different kinds of ModelEntity implenentations, it would make sense to use a "real" equals method, wdyt? (also please use { } for this condition, it's clearer that in one line)

Reply:
Yeah, never use `!=` for something different than primitives